### PR TITLE
replace container-interop/container-interop with psr/container

### PR DESCRIPTION
**Pull request by 1ma**
## Abstract

`container-interop/container-interop` is deprecated since Feb. 13th 2017 in favor of `psr/container` (PSR-11 spec). This PR replaces one for the other.

### My use case

I use Pimple as the DI container to lazy load command handlers. Pimple provides it's own [PSR-11 decorator](https://github.com/silexphp/Pimple/blob/master/src/Pimple/Psr11/Container.php), but it implements PSR-11's interface, not container-interop's, hence it cannot be fed to the `Dispatcher` class.

### Promotion of `psr/container` to regular dependency

The `Dispatcher` class, which lives under `src` uses it, so I believe it should not be disguised as a development/suggested depedency (if it's not installed could break production code).

### Potential BC breaks?

Since release 1.2.0 of container-interop `Interop\Container\ContainerInterface` extends `Psr\Container\ContainerInterface`, so I understand that there should be none: interop containers are also psr-11 containers (but the inverse does not hold).

### ~~About the removal of composer.lock~~

~~When installing a dependency with composer the composer.lock file is ignored, and most PHP libraries keep it out of the repository because it is just dead weight.~~

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
+composer.lock
```
Comment (webimpress) [inline .gitignore]: It should not be ignored, we'd like to run test with locked dependencies (see Travis CI configuration in .travis.yml)

Comment (1ma) [reply .gitignore]: Got it!

**Comment by webimpress** [comment]
@1ma It's more complicated to fix these tests here (will be out of the scope). See my PR #41 
I think when #41 will be merged then your PR should work fine

**Comment by 1ma** [comment]
Great! Once your PR is merged into develop I'll rebase mine and see what happens.

**Comment by webimpress** [comment]
@1ma meanwhile you can try on local to make sure all is good :smile:

**Comment by 1ma** [comment]
Myea, success.

**Comment by weierophinney** [comment]
Thanks, @1ma

**Pull request merged**